app/models/time_tracker_extension/time_locking_period.rb

```diff
@@ -13,18 +13,12 @@ def unblock!
       related_periods.update_all(approved: false)
     end
 
-    def approve!
-      return if approved
-      update(approved: true)
-      TimeTrackerExtension::SendPeriodReportsJob.perform_later(self)
-    end
-
     private
 
     def does_not_contain_running_tasks
-      return unless approved
       if workspace.time_records.
```

Comment:
I think lines 19-23 should be in some instance method like #contains_running_tasks?